conda_build/build.py

```diff
@@ -1347,24 +1347,39 @@ def _construct_metadata_for_test_from_package(package, config):
             config.filename_hashing = False
             hash_input = {}
 
-    local_location = os.path.dirname(package)
-    # strip off extra subdir folders
-    for platform in ('win', 'linux', 'osx'):
-        if os.path.basename(local_location).startswith(platform + "-"):
-            local_location = os.path.dirname(local_location)
+    log = utils.get_logger(__name__)
 
-    if not os.path.abspath(local_location):
-        local_location = os.path.normpath(os.path.abspath(
-            os.path.join(os.getcwd(), local_location)))
+    # get absolute file location
+    local_pkg_location = os.path.normpath(os.path.abspath(os.path.dirname(package)))
+
+    # get last part of the path
+    last_element = os.path.basename(local_pkg_location)
+    is_channel = False
+    for platform in ('win-', 'linux-', 'osx-', 'noarch'):
+        if last_element.startswith(platform):
+            is_channel = True
+
+    if not is_channel:
+        log.error("Your package must reside in a channel structure with platform-subfolders!")
+        sys.exit(1)
```

Comment:
here I'm not sure how to properly throw an error?!

Reply:
```raise ValueError("Your package must reside in a channel structure with platform-subfolders!")```

A more specific exception type might be nice, but ValueError is probably fine to start with.

Reply:
done...